File: Java/src/session3/td/ToUpperCase.java
package session3.td;

public class ToUpperCase implements Visiteur<String> {
	
	@Override
	public String casVide() {
		return "";
	}

	@Override
	public String casCons(char lettre, String resteMot) {
		String upperCaseStr = Character.toUpperCase(lettre) + resteMot.toUpperCase();
		return upperCaseStr;
	}

}

File: TD3/typescript/session3/td/ToUpperCase.ts
import { Visiteur } from "./Visiteur";

export class ToUpperCase implements Visiteur<string>{
    casCons(l: string, r: string): string {
        if(l!=undefined){
            var upperCaseStr: string = l.toUpperCase() + r.toUpperCase();
        return upperCaseStr;
        } else {
            return "";
        }
    }
    casVide(): string {
        return "";
    }


}